bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/v18/V18AssortedRpc.scala

```diff
@@ -0,0 +1,48 @@
+package org.bitcoins.rpc.client.v18
+import org.bitcoins.core.protocol.blockchain.BlockHeader
+import org.bitcoins.rpc.client.common.Client
+import org.bitcoins.rpc.jsonmodels.{
+  GetNodeAddressesResult,
+  GetRpcInfoResult,
+  ListWalletDirResult
+}
+import org.bitcoins.rpc.serializers.JsonSerializers._
+import play.api.libs.json.{JsString, Json}
+
+import scala.concurrent.Future
+
+/**
+  * Assorted Rpc calls for Bitcoin V18
+  * @see [[https://bitcoincore.org/en/doc/0.18.0/rpc/network/getnodeaddresses/]]
+  * @see [[https://bitcoincore.org/en/doc/0.18.0/rpc/wallet/listwalletdir/]]
+  * @see [[https://github.com/bitcoin/bitcoin/commit/e82f6ad6f270f1f101d8853be32fd11eff4ddfb8]]
+  * @see [[https://bitcoincore.org/en/doc/0.18.0/rpc/mining/submitheader/]]
+  */
+trait V18AssortedRpc {
+  self: Client =>
+
+  private def getNodeAddresses(
+      count: Option[Int]): Future[Vector[GetNodeAddressesResult]] = {
+    bitcoindCall[Vector[GetNodeAddressesResult]]("getnodeaddresses",
+                                                 List(Json.toJson(count)))
```

Comment:
In the `None` case this turns into `"null"`, is this what you want? Or should you just implement the two public `getNodeAddresses` calls individually?

Reply:
Torkel suggested I write it this way to have all possible cases I believe. If the two calls individually work better I can implement that